src/parseMessageDefinition.js

```diff
@@ -190,20 +193,28 @@ export function parseMessageDefinition(messageDefinition: string) {
     .map((line) => line.trim())
     .filter((line) => line);
 
-  let definitionLines: string[] = [];
+  let definitionLines: { isJson: boolean, line: string }[] = [];
   const types: RosMsgDefinition[] = [];
+  let nextDefinitionIsJson: boolean = false;
   // group lines into individual definitions
   allLines.forEach((line) => {
-    // skip comment lines
-    if (line.indexOf("#") === 0) {
+    // ignore comment lines unless they start with #pragma rosbag_parse_json
+    if (line.indexOf("#") === 0 && line.indexOf("#pragma rosbag_parse_json") !== 0) {
```

Comment:
is the `#` guaranteed to be at index 0? I assume these definitions don't have any prefixed whitespace chars?

Reply:
We trim leading and trailing whitespace chars, so it should be guaranteed to be at index 0, if the line is truly a pragma comment.